experimental/sdk/webpubsub-socketio-extension/src/SIO/index.ts

```diff
@@ -48,13 +54,21 @@ export async function useAzureSocketIOChain(
   // current listeners = EIO handleRequest listeners (e.g. /socket.io) + other listeners from user
   const listeners = httpServer.listeners("request").slice(0);
   httpServer.removeAllListeners("request");
-  let nativeServiceClient: WebPubSubServiceClient;
+
+  this[WEB_PUBSUB_OPTIONS_PROPERY_NAME] = webPubSubOptions;
+  if (webPubSubOptions.configureNegotiateOptions) {
+    if (this[NEGOTIATE_HANDLER_PROPERTY_NAME]) throw new Error("Negotiate options has already been set.");
+    else
+      this[NEGOTIATE_HANDLER_PROPERTY_NAME] = getNegotiateHandler(
+        webPubSubOptions,
+        webPubSubOptions.configureNegotiateOptions
+      );
+  }
+
   httpServer.on("request", (req: IncomingMessage, res: ServerResponse) => {
     // negotiate handler
-    if (webPubSubOptions.configureNegotiateOptions && checkNegotiate(req.url)) {
-      nativeServiceClient = getWebPubSubServiceClient(webPubSubOptions);
-      const negotiateHandler = getNegotiateHandler();
-      negotiateHandler(req, res, webPubSubOptions.configureNegotiateOptions, nativeServiceClient);
+    if (this[NEGOTIATE_HANDLER_PROPERTY_NAME] && checkNegotiate(req.url)) {
```

Comment:
I have a concern that does it mean negotiate always be called in the beginning of middleware?

Reply:
yes. 
We have to check whether negotiate handler is existing for every request.
Because it could be added after `useAzuresocketIO` by using our new middleware.

Reply:
Can it be a problem? Say user use passport to auth with 3rd party like Gtihub, does it work in such scenario? I think passport's middleware needs to be called first?